static/js/invite.js

```diff
@@ -270,73 +236,128 @@ function set_custom_time_inputs_visibility() {
+            $("#custom_expires_on").text(valid_to(get_expiration_time_in_minutes()));
+        });
+
+        $("#invite_check_all_button").on("click", () => {
+            $("#streams_to_add :checkbox").prop("checked", true);
+            toggle_invite_submit_button();
+        });
+
+        $("#invite_uncheck_all_button").on("click", () => {
+            $("#streams_to_add :checkbox").prop("checked", false);
+            $("#invite-user-modal .dialog_submit_button").prop(
+                "disabled",
+                !$("#generate_multiuse_invite_radio").is(":checked"),
+            );
+        });
+    }
+
+    function invite_users() {
         const is_generate_invite_link = $("#generate_multiuse_invite_radio").prop("checked");
         if (is_generate_invite_link) {
             generate_multiuse_invite();
         } else {
             submit_invitation_form();
         }
-    });
-
-    $("#generate_multiuse_invite_button").on("click", () => {
-        $("#generate_multiuse_invite_radio").prop("checked", true);
-        $("#multiuse_radio_section").show();
-        $("#invite-method-choice").hide();
-        $("#invitee_emails").prop("disabled", true);
-        $("#submit-invitation").text($t({defaultMessage: "Generate invite link"}));
-        $("#submit-invitation").data("loading-text", $t({defaultMessage: "Generating link..."}));
-        reset_error_messages();
-    });
-
-    $("#invite-user").on("change", "#generate_multiuse_invite_radio", () => {
-        $("#invitee_emails").prop("disabled", false);
-        $("#submit-invitation").text($t({defaultMessage: "Invite"}));
-        $("#submit-invitation").data("loading-text", $t({defaultMessage: "Inviting..."}));
-        $("#multiuse_radio_section").hide();
-        $("#invite-method-choice").show();
-        reset_error_messages();
-    });
-
-    $("#expires_on").text(valid_to($("#expires_in").val()));
-    $("#expires_in").on("change", () => {
-        set_custom_time_inputs_visibility();
-        set_expires_on_text();
-    });
+    }
 
-    $(".custom-expiration-time").on("change", () => {
-        custom_expiration_time_input = $("#custom-expiration-time-input").val();
-        custom_expiration_time_unit = $("#custom-expiration-time-unit").val();
-        $("#custom_expires_on").text(valid_to(get_expiration_time_in_minutes()));
+    dialog_widget.launch({
+        html_heading: $t_html({defaultMessage: "Invite users to Zulip"}),
+        html_body,
+        html_submit_button: $t_html({defaultMessage: "Invite"}),
+        id: "invite-user-modal",
+        loading_spinner: true,
+        on_click: invite_users,
+        post_render: invite_user_modal_post_render,
     });
+}
 
-    $("#custom-expiration-time-input").on("keydown", (e) => {
-        if (keydown_util.is_enter_event(e)) {
-            e.preventDefault();
-            return;
-        }
-    });
+export function initialize() {
+    $(document).on("click", ".invite-user-link", open_invite_user_modal);
```

Comment:
Sould this be a proper click handler that does `preventDefault/stopPropagation`?

Reply:
The `open_invite_user_modal` function does make [calls](https://github.com/zulip/zulip/pull/23466/files#diff-dc7103df355c47e67a9af54d3a6de32f3d91cf8949911a0cd976fe73c907c610R240-R241) to `preventDefault/stopPropagation`